crates/ursa-store/src/store.rs

```diff
@@ -84,6 +87,58 @@ pub trait BlockstoreExt: Blockstore {
 
         Ok(cids)
     }
+
+    /// traverse a dag and get full dag given a root cid
+    fn dag_traversal(&self, root_cid: &Cid) -> Result<Vec<(Cid, Vec<u8>)>> {
```

Comment:
Moving this here is not appropriate. This a blockstore extension, aimed at 

> /// Extension methods for inserting and retrieving IPLD data with CIDs

Reply:
Wouldn't traversing a dag be `retrieving IPLD data with CIDs` ? Was suggested to move here:  https://github.com/fleek-network/ursa/pull/278#discussion_r1081651261

Reply:
I can move it directly into UrsaStore, but figured it'd be more generic and usable if on the BlockstoreExt instead, as all you need is an inner store for the methods

Reply:
> Wouldn't traversing a dag be `retrieving IPLD data with CIDs` ? Was suggested to move here: [#278 (comment)](https://github.com/fleek-network/ursa/pull/278#discussion_r1081651261)

I suggested that we move these to `UrsaStore` not `BlockstoreExt`.

Reply:
> I can move it directly into UrsaStore, but figured it'd be more generic and usable if on the BlockstoreExt instead, as all you need is a store for the methods

What helps me decide whether a method should go in a trait is "does it make sense for all users of trait X to have to implement this behaviour/method?".

Reply:
https://github.com/fleek-network/ursa/pull/278/commits/582dec7c6012b72a74513978db247b00184c9bdb